File: test-server.js
const http = require('http');

const server = http.createServer((req, res) => {
    console.log(`${req.method} ${req.url}`);
    
    if (req.method === 'GET' && req.url === '/status') {
        res.writeHead(200, { 'Content-Type': 'application/json' });
        res.end(JSON.stringify({
            status: 'running',
            server: 'Node.js Test Server',
            timestamp: new Date().toISOString()
        }));
    }
    else if (req.method === 'POST' && req.url === '/emergency') {
        let body = '';
        req.on('data', chunk => body += chunk);
        req.on('end', () => {
            console.log('Emergency received:', body);
            res.writeHead(200, { 'Content-Type': 'application/json' });
            res.end(JSON.stringify({
                status: 'received',
                message: 'Emergency signal processed'
            }));
        });
    }
    else {
        res.writeHead(200, { 'Content-Type': 'text/html' });
        res.end(`
            <html>
            <body>
                <h1>Test Emergency Server (Node.js)</h1>
                <p>Server is running on port 8080</p>
                <p>Endpoints available:</p>
                <ul>
                    <li><strong>POST /emergency</strong> - For ESP-01S emergency</li>
                    <li><strong>GET /status</strong> - Server status check</li>
                </ul>
                <p>Current time: ${new Date().toString()}</p>
            </body>
            </html>
        `);
    }
});

server.listen(8080, () => {
    console.log('Test server running on http://localhost:8080');
    console.log('Ready for ESP-01S connections');
});

File: test_server.py
from http.server import HTTPServer, BaseHTTPRequestHandler
import json
import time

class TestHandler(BaseHTTPRequestHandler):
    def do_GET(self):
        if self.path == '/status':
            self.send_response(200)
            self.send_header('Content-type', 'application/json')
            self.end_headers()
            response = {
                "status": "running",
                "server": "Python Test Server",
                "time": time.strftime("%Y-%m-%d %H:%M:%S"),
                "message": "Ready for ESP-01S connections"
            }
            self.wfile.write(json.dumps(response).encode())
        
        elif self.path == '/':
            self.send_response(200)
            self.send_header('Content-type', 'text/html')
            self.end_headers()
            html = """
            <html>
            <body>
                <h1>Test Emergency Server</h1>
                <p>Server is running!</p>
                <p>Endpoints:</p>
                <ul>
                    <li>POST /emergency - For ESP-01S emergency signal</li>
                    <li>GET /status - Check server status</li>
                </ul>
                <p>Time: """ + time.strftime("%Y-%m-%d %H:%M:%S") + """</p>
            </body>
            </html>
            """
            self.wfile.write(html.encode())
    
    def do_POST(self):
        if self.path == '/emergency':
            content_length = int(self.headers['Content-Length'])
            post_data = self.rfile.read(content_length)
            
            print(f"Received emergency: {post_data.decode()}")
            
            self.send_response(200)
            self.send_header('Content-type', 'application/json')
            self.end_headers()
            response = {
                "status": "received",
                "message": "Emergency signal received",
                "timestamp": time.time()
            }
            self.wfile.write(json.dumps(response).encode())

def run_server():
    server_address = ('', 8080)  # Listen on all interfaces, port 8080
    httpd = HTTPServer(server_address, TestHandler)
    print(f"Test server running on http://localhost:8080")
    print(f"Access from ESP-01S using your PC's IP")
    print(f"Check your PC IP with: ipconfig (Windows) or ifconfig (Linux/Mac)")
    httpd.serve_forever()

if __name__ == '__main__':
    run_server()